streams/src/api/user.rs

```diff
@@ -213,12 +255,19 @@ impl<T> User<T> {
         // Store message content into stores
         let author_id = message.payload().content().author_id();
         let author_ke_pk = message.payload().content().author_ke_pk();
-        self.state.exchange_keys.insert(author_id, author_ke_pk);
-        self.state.stream_address = Some(address);
+        if is_base_branch {
+            self.state.exchange_keys.insert(author_id, author_ke_pk);
+            self.state.base_topic = topic.clone();
+            self.state.stream_address = Some(address);
+        }
+        // Update branch links
+        self.set_anchor(&topic, address)?;
+        self.set_latest_link(&topic, address)?;
         self.state.author_identifier = Some(author_id);
```

Comment:
you can avoid a `topic.clone()` just by reordering the steps:

```suggestion
        self.set_anchor(&topic, address)?;
        self.set_latest_link(&topic, address)?;
        self.state.author_identifier = Some(author_id);
        if is_base_branch {
            self.state.exchange_keys.insert(author_id, author_ke_pk);
            self.state.base_topic = topic;
            self.state.stream_address = Some(address);
        }
```